// oc14-modal-library/dist/index.modern.js
import React, { Fragment } from 'react';

var backdropStyles = {
  position: "fixed",
  top: "0",
  left: "0",
  width: "100%",
  height: "100vh",
  backgroundColor: "rgba(0, 0, 0, .8)",
  zIndex: "5"
};
var modalHeaderButtonStyle = {
  display: "block",
  border: "none",
  background: "transparent",
  width: "auto",
  margin: "0",
  padding: "0",
  marginLeft: "auto",
  position: "fixed",
  top: "1rem",
  right: "1rem",
  zIndex: "6"
};
var modalHeaderIconStyle = {
  width: "1rem",
  fill: "white"
};

var Backdrop = function Backdrop(_ref) {
  var onClick = _ref.onClick,
      onClickCloseBtn = _ref.onClickCloseBtn;
  return /*#__PURE__*/React.createElement("div", {
    style: backdropStyles,
    onClick: onClick
  }, /*#__PURE__*/React.createElement("button", {
    style: modalHeaderButtonStyle,
    onClick: onClickCloseBtn
  }, /*#__PURE__*/React.createElement("svg", {
    style: modalHeaderIconStyle,
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 320 512"
  }, /*#__PURE__*/React.createElement("path", {
    d: "M310.6 361.4c12.5 12.5 12.5 32.75 0 45.25C304.4 412.9 296.2 416 288 416s-16.38-3.125-22.62-9.375L160 301.3L54.63 406.6C48.38 412.9 40.19 416 32 416S15.63 412.9 9.375 406.6c-12.5-12.5-12.5-32.75 0-45.25l105.4-105.4L9.375 150.6c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0L160 210.8l105.4-105.4c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25l-105.4 105.4L310.6 361.4z"
  }))));
};

var modalContentStyle = {
  width: "100%"
};

var Modal = function Modal(_ref) {
  var children = _ref.children,
      show = _ref.show,
      onClickCloseBtn = _ref.onClickCloseBtn;
  return /*#__PURE__*/React.createElement(Fragment, null, show && /*#__PURE__*/React.createElement(Fragment, null, /*#__PURE__*/React.createElement(Backdrop, {
    onClick: onClickCloseBtn
  }), /*#__PURE__*/React.createElement("div", {
    style: modalContentStyle
  }, children)));
};

export default Modal;
